test/parallel/test-module-binding.js

```diff
@@ -4,34 +4,14 @@ require('../common');
 const fixtures = require('../common/fixtures');
 const { internalBinding } = require('internal/test/binding');
 const { internalModuleReadJSON } = internalBinding('fs');
-const { readFileSync } = require('fs');
 const { strictEqual } = require('assert');
+
 {
-  const [string, containsKeys] = internalModuleReadJSON('nosuchfile');
-  strictEqual(string, undefined);
-  strictEqual(containsKeys, undefined);
+  strictEqual(internalModuleReadJSON('nosuchfile'), undefined);
 }
 {
-  const [string, containsKeys] =
-    internalModuleReadJSON(fixtures.path('empty.txt'));
-  strictEqual(string, '');
-  strictEqual(containsKeys, false);
+  strictEqual(internalModuleReadJSON(fixtures.path('empty.txt')), '');
 }
 {
-  const [string, containsKeys] =
-    internalModuleReadJSON(fixtures.path('empty.txt'));
-  strictEqual(string, '');
-  strictEqual(containsKeys, false);
-}
-{
-  const [string, containsKeys] =
-    internalModuleReadJSON(fixtures.path('empty-with-bom.txt'));
-  strictEqual(string, '');
-  strictEqual(containsKeys, false);
-}
-{
-  const filename = fixtures.path('require-bin/package.json');
-  const [string, containsKeys] = internalModuleReadJSON(filename);
-  strictEqual(string, readFileSync(filename, 'utf8'));
-  strictEqual(containsKeys, true);
+  strictEqual(internalModuleReadJSON(fixtures.path('empty-with-bom.txt')), '');
```

Comment:
We are no longer testing JSON files? Why? That looks like a mistake.

Reply:
Is there some test that validates that `internalModuleReadJSON` is actually able to read JSON?

Reply:
How do you validate if a read operation is done? If theall tests pass until this test is executed it 100% validates in-directly that this function is executed correctly. Therefore, we validate the read operation by not failing until this test is executed. I can add a test that somehow validates output, but it will be useless because if that test breaks, we would know it failed before we even run the test. I hope that makes sense.

Reply:
I don; t think redundancy in tests is useless, on the contrary it can be quite useful to have such test to understand where the failure comes from 🤷‍♂️

Reply:
> I can add a test that somehow validates output, but it will be useless because if that test breaks, we would know it failed before we even run the test. I hope that makes sense.

Mm, I see where you're coming from, but that's how integration tests work: if the pieces they consume break, the integration breaks. The value of the integration test is that you know the particular combination and configuration of the smaller pieces results in the specific outcome desired.

If there is no-longer a test covering this function, that says to me either the function is unnecessary or coverage is missing.

If there's merely orchestration of complicated deps happening, perhaps mock them and assert this function's output is correct. That avoids the cascading breakage you cited.

Reply:
This test is covered by virtue of our tests that a specifier loads a package’s `"main"` etc. I don’t think we need a more specific test to cover the `package.json`-reading that happens within that operation.

Reply:
I agree with your statements, but strongly disagree with adding a test for this case, since this function is the core of module loading, but @aduh95 and @JakobJingleheimer's opinions are important for me, and added the test as a new commit. Please resolve this conversation, if you think the test is sufficient. Thank you for your time and your review.

Reply:
I agree with your statements, but strongly disagree with adding a test for this case, since this function is the core of module loading, but all of your opinions are important for me, and added the test as a new commit. Please resolve this conversation, if you think the test is sufficient. Thank you for your time and your review.

Reply:
Thanks for adding the test, it makes it possible to understand the function intent without reading the source code, and that's certainly a good addition so reviewers who are not fluent in C++ can be more confident in what the function is doing ❤️